src/field.cpp

```diff
@@ -1836,58 +1830,71 @@ void map::player_in_field( player &u )
                 }
             }
             {
+                // Burn message by intensity
+                static const std::array<std::string, 4> player_burn_msg = {{
+                    _("You burn your legs and feet!"),
+                    _("You're burning up!"),
+                    _("You're set ablaze!"),
+                    _("Your whole body is burning!")
+                }};
+                static const std::array<std::string, 4> npc_burn_msg = {{
+                    _("<npcname> burns their legs and feet!"),
+                    _("<npcname> is burning up!"),
+                    _("<npcname> is set ablaze!"),
+                    _("<npcname>s whole body is burning!")
+                }};
+                static const std::array<std::string, 4> player_warn_msg = {{
+                    _("You're standing in a fire!"),
+                    _("You're waist-deep in a fire!"),
+                    _("You're surrounded by raging fire!"),
+                    _("You're lying in fire!")
+                }};
+
+                int burn_min = adjusted_intensity;
+                int burn_max = 3 * adjusted_intensity + 3;
                 std::list<int> parts_burned;
```

Comment:
While you're here, you could change this to be a vector of `body_part` and get rid of the cast at 1884.